packages/number-input/src/use-number-input.ts

```diff
@@ -453,9 +452,7 @@ export function useNumberInput(props: UseNumberInputProps = {}) {
       autoCorrect: "off",
       onChange: callAllHandlers(props.onChange, onChange),
       onKeyDown: callAllHandlers(props.onKeyDown, onKeyDown),
-      onFocus: callAllHandlers(props.onFocus, onFocusProp, () =>
-        scheduleMicrotask(setFocused.on),
-      ),
+      onFocus: callAllHandlers(props.onFocus, onFocusProp, () => setFocused.on),
```

Comment:
I think we need to pass the function definition here, not a new callback which returns the setter.

```suggestion
      onFocus: callAllHandlers(props.onFocus, onFocusProp, setFocused.on),
```

Reply:
Oh yes, thank you for catching that!